backend/inspirehep/assign/views.py

```diff
@@ -78,8 +78,10 @@ def author_assign_view():
     literature_recids = body["literature_recids"]
     with db.session.begin_nested():
         if to_author_recid is None:
+            from_author = AuthorsRecord.get_record_by_pid_value(from_author_recid)
+            from_author_name = from_author["name"]["value"]
```

Comment:
I don't think this is the name to use. Basically you need to take the logic from `update_author_names` and instead of mutating the already constructed author, it should provide you with the author name, that you can then pass to `create_new_stub_author`. I don't think you need to change anything outside `assign_to_new_stub_author`.

Reply:
Ok, but the name used to create name vairants (as in `update_author_names`) should be the name of the existing author that we're unassigning records from?